src/plugins/orangehrmRecruitmentPlugin/Controller/ApplicantController.php

```diff
@@ -0,0 +1,317 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Recruitment\Controller;
+
+use Exception;
+use OrangeHRM\Authentication\Csrf\CsrfTokenManager;
+use OrangeHRM\Authentication\Exception\AuthenticationException;
+use OrangeHRM\Core\Api\V2\Exception\InvalidParamException;
+use OrangeHRM\Core\Api\V2\Validator\Helpers\ValidationDecorator;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Api\V2\Validator\ValidatorException;
+use OrangeHRM\Core\Controller\AbstractController;
+use OrangeHRM\Core\Controller\PublicControllerInterface;
+use OrangeHRM\Core\Dto\Base64Attachment;
+use OrangeHRM\Core\Traits\LoggerTrait;
+use OrangeHRM\Core\Traits\ORM\EntityManagerHelperTrait;
+use OrangeHRM\Core\Traits\Service\DateTimeHelperTrait;
+use OrangeHRM\Core\Traits\Service\NormalizerServiceTrait;
+use OrangeHRM\Core\Traits\ValidatorTrait;
+use OrangeHRM\Entity\Candidate;
+use OrangeHRM\Entity\CandidateAttachment;
+use OrangeHRM\Entity\CandidateHistory;
+use OrangeHRM\Entity\CandidateVacancy;
+use OrangeHRM\Entity\WorkflowStateMachine;
+use OrangeHRM\Framework\Http\Request;
+use OrangeHRM\Framework\Http\Response;
+use OrangeHRM\ORM\Exception\TransactionException;
+use OrangeHRM\Recruitment\Service\CandidateService;
+use OrangeHRM\Recruitment\Traits\Service\CandidateServiceTrait;
+use OrangeHRM\Recruitment\Traits\Service\RecruitmentAttachmentServiceTrait;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+class ApplicantController extends AbstractController implements PublicControllerInterface
+{
+    use CandidateServiceTrait;
+    use RecruitmentAttachmentServiceTrait;
+    use ValidatorTrait;
+    use EntityManagerHelperTrait;
+    use NormalizerServiceTrait;
+    use LoggerTrait;
+    use DateTimeHelperTrait;
+
+    public const PARAMETER_FIRST_NAME = 'firstName';
+    public const PARAMETER_MIDDLE_NAME = 'middleName';
+    public const PARAMETER_LAST_NAME = 'lastName';
+    public const PARAMETER_EMAIL = 'email';
+    public const PARAMETER_RESUME = 'resume';
+    public const PARAMETER_CONTACT_NUMBER = 'contactNumber';
+    public const PARAMETER_VACANCY_ID = 'vacancyId';
+    public const PARAMETER_CONSENT_TO_KEEP_DATA = 'consentToKeepData';
+
+    /**
+     * @var ValidationDecorator|null
+     */
+    private ?ValidationDecorator $validationDecorator = null;
+
+    /**
+     * @throws AuthenticationException
+     * @throws ValidatorException
+     * @throws TransactionException
+     */
+
+    public function handle(Request $request): Response
```

Comment:
lengthly method. can split to private methods

Reply:
Done